### ISSUE TITLE
sql: TestSteady index backfill seems to not get run properly

### ISSUE THREAD

**Issue by jordanlewis** [cockroachdb/cockroach#15038]
See the logs in https://teamcity.cockroachdb.com/viewLog.html?buildId=210435&buildTypeId=Cockroach_Nightlies_Steady_6Medium&tab=artifacts

```
I170406 21:17:23.892342 343 acceptance/allocator_test.go:234  starting schema change: ALTER TABLE datablocks.blocks ADD COLUMN newcol DECIMAL DEFAULT (DECIMAL '1.4')
I170406 21:17:35.097040 344 acceptance/allocator_test.go:239  completed schema change: CREATE INDEX foo ON datablocks.blocks (block_id), in 11.205213575s
```

How could the `CREATE INDEX` have only taken 11 seconds over 57M rows?

**Comment by vivekmenezes** [comment]
This issue is caused by a schema changer not seeing its id at the start of the mutation queue in a table descriptor. This happens when two schema changes are queued up at around the same time with one ahead of the other. Since it's not at the start of the mutation queue it declares itself done as far as the synchronous execution of the schema change is concerned and leaves itself to be run by the asynchronous path.

So while it hasn't yet completed, the sql command itself returns to the user with no error. We should fix this to have it wait until the schema change is complete.

**Comment by cuongdo** [comment]
Does this need to be documented, as a release note? This seems counterintuitive. As a user, I'd expect a command to either return immediately (if it launches something async) or after the operation has finished.

**Comment by vivekmenezes** [comment]
This happens only when a user runs two schema changes at the same time. I'm
okay with adding it to the release notes, but I doubt it's at the top of
the list of issues. We should certainly fix it in 1.1
